projects/observability/src/pages/apis/service-detail/instrumentation/service-instrumentation.service.ts

```diff
@@ -11,12 +12,25 @@ export class ServiceInstrumentationService {
     ServiceScoreResponse | undefined
   >(undefined);
 
-  public getServiceScore(serviceName: string): Observable<ServiceScoreResponse> {
-    return of({ ...serviceScoreResponse, serviceName: serviceName });
+  public constructor(private readonly queryService: InstrumentationQualityService) {}
+
+  public getServiceScore(
+    serviceName: string,
+    options: { mock: boolean } = { mock: false }
+  ): Observable<ServiceScoreResponse> {
+    if (options.mock) {
+      return of({ ...serviceScoreResponse, serviceName: serviceName }).pipe(delay(1000));
+    }
+
+    return this.queryService.get<ServiceScoreResponse>(`/${serviceName}`);
   }
 
-  public getOrgScore(): Observable<OrgScoreResponse> {
-    return of(orgScoreResponse);
+  public getOrgScore(options: { mock: boolean } = { mock: false }): Observable<OrgScoreResponse> {
```

Comment:
A bit against the mock pattern. I feel it might not be removed in tree shaking and our bundles might include mocks here. Can you check once?

Reply:
Added the mock option to quickly switch between backend and mock values to experiment with different scenarios. Ideally, the mocks shouldn't be required anymore. 

Should I remove the mock option from the method? Not sure how to remove it from the bundle.

Reply:
Also, it might be a common use-case to see how the UI looks based on different backend responses. Like [this edge case][1]. Unless we spin up a local mock server, what's the recommended approach to mocking the backend response to quickly view different scenarios?

[1]: https://razorpay.slack.com/archives/CU5GKS8MQ/p1664527835735519?thread_ts=1664523417.972659&cid=CU5GKS8MQ

Reply:
I have some commented lines of code, where in in the `catchError`, I'm returning an observable and then adding a `defaultIfEmpty` block. Of course, will push these lines to codebase, but they'll be commented. Any time I want to leverage any response, I can just add the JSON and import the file.

Reply:
✅ Commented the code that returns mock response and removed the `options` parameter.